full-moon/src/ast/parsers.rs

```diff
@@ -789,11 +789,31 @@ define_parser!(ParseFunctionBody, FunctionBody<'a>, |_, state| {
                 parameters.push(last_parameter);
 
                 parameters.push(Pair::new(Parameter::Ellipse(ellipse), None));
+
+                // Parse ellipse type if in Luau
+                if cfg!(feature = "roblox") {
```

Comment:
This needs to be `cfg_if`.

This is going to compile to `if false`, but the code inside it will still attempt to compile. `ParserTypeSpecifier` and friends don't (or rather, shouldn't), exist without the `roblox` feature flag, so this won't compile.

Reply:
For whatever reasons the tests say this is fine, which means `ParseTypeSpecifier` is for whatever reason on non-roblox builds.

This should still probably be cfg_if.

Reply:
I did not change this from how it was previously done, so yeah ParseTypeSpecifier was present in non-roblox builds (I think this is what the `define_roblox_parser!` macro is for? It falls back to a noop which will never run, parsing a TokenReference in non roblox builds)

I can change this to cfg_if though

Reply:
Oh, fair enough, I forgot.